kedro/extras/datasets/plotly/json_dataset.py

```diff
@@ -27,7 +27,20 @@ class JSONDataSet(
     """``JSONDataSet`` loads/saves a plotly figure from/to a JSON file using an
     underlying filesystem (e.g.: local, S3, GCS).
 
-    Example:
+    Example adding a catalog entry with
+    `YAML API
+    <https://kedro.readthedocs.io/en/stable/data/\
+        data_catalog.html#use-the-data-catalog-with-the-yaml-api>`_:
```

Comment:
I am unsure about having hardcoded URLs in our API docs. Ideally I'd prefer to avoid this and I think eventually we will move those examples **out** of the main docs and fully into the API docs. 

Basically, you've beaten us to it -- you've done the right thing to move the example out to here, and we will do the same, so I don't think you need to link back to the docs page!

Reply:
Generally agree with this one. For this case, if the example is moved into the Class itself already, I don't think it's necessary to keep that link (Which is nice since you can see it from the IDE hints or hit `Shift + Tab` on Jupyter to see the example immediately)

There are some cases hard-coded URL is still useful - for example, linking to `pandas` for the full argument list etc.